templates/decl.j2

```diff
@@ -54,6 +54,17 @@
     </details>
 {% endif %}
 
+{% if decl.name in instances_for %}
+    <details class="instances-for">
+        <summary>Available instances</summary>
```

Comment:
I found the terminology a bit confusing.  We have "instances" and "available instances", but I couldn't tell you which is which.  maybe we should look at how the haskell documentation calls this.  Maybe "instances of this type class" and "implements the following type classes"?

Reply:
Haskell seems to use "instances" for both versions:

* https://hackage.haskell.org/package/base-4.16.1.0/docs/Foreign-C-Types.html#t:CBool (instances on a type)
* https://hackage.haskell.org/package/base-4.16.1.0/docs/Prelude.html#t:Integral (instances of a typeclass)

The reality is that probably we never show both these headings anyway

Reply:
> The reality is that probably we never show both these headings anyway

Actually we do: https://leanprover-community.github.io/mathlib_docs_demo/algebra/group/defs.html#group

Reply:
Oh, nice find.

Reply:
Let me know if the new logic looks better